diff --git a/vllm_ascend/worker/v2/model_runner.py b/vllm_ascend/worker/v2/model_runner.py
--- a/vllm_ascend/worker/v2/model_runner.py
+++ b/vllm_ascend/worker/v2/model_runner.py
@@ -50,7 +50,8 @@ class NPUModelRunner(GPUModelRunner):
     def __init__(self, vllm_config: VllmConfig, device: torch.device):
         with torch_cuda_wrapper():
             super().__init__(vllm_config, device)
 
+        cache_draft_logits = self.req_states.draft_logits is not None
         # because we will override these attribute, delete these attribute to
         # make sure it's collected by python gc immediately.
         del self.cudagraph_manager
@@ -83,6 +84,8 @@ class NPUModelRunner(GPUModelRunner):
             num_speculative_steps=self.num_speculative_steps,
             vocab_size=self.vocab_size,
             device=self.device,
+            model_dtype=self.dtype,
+            cache_draft_logits=cache_draft_logits,
         )
         # AscendInputBuffers has extra `seq_lens_cpu` attribute.
         # so reinitialize input_buffers here.
diff --git a/vllm_ascend/worker/v2/states.py b/vllm_ascend/worker/v2/states.py
--- a/vllm_ascend/worker/v2/states.py
+++ b/vllm_ascend/worker/v2/states.py
@@ -31,6 +31,8 @@ class AscendRequestState(RequestState):
         num_speculative_steps: int,
         vocab_size: int,
         device: torch.device,
+        model_dtype: torch.dtype,
+        cache_draft_logits: bool,
     ):
         super().__init__(
             max_num_reqs,
@@ -39,6 +41,8 @@ class AscendRequestState(RequestState):
             num_speculative_steps,
             vocab_size,
             device,
+            model_dtype,
+            cache_draft_logits,
         )
         # vllm gpu_model_runner_v2 deprecate the seqs_lens_cpu attribute,
         # because they think most attention backends do not need it.
